packages/prerender-proxy/lib/handlers/prerender-check.ts

```diff
@@ -15,12 +15,6 @@ export const handler = async (
   // then set the x-request-prerender header so the origin-request lambda function
   // alters the origin to prerender.io
   if (IS_BOT.test(request.headers["user-agent"][0].value)) {
```

Comment:
On a side note I'm wondering whether this function is needed at all.

CloudFront behaviour can just simply forward all headers which means this function is not necessary at the viewer request right.

Simply move the existing logic in `prerender.ts` under this condition and we save invoking the viewer request lambda?

Reply:
@AdamJHall @TheOrangePuff @crispy101 Thoughts?

Reply:
Is the user agent part of the cache key though? Otherwise we would end up serving the prerender response to everyone.

Reply:
True but that would be controlled outside of this construct wouldn't it?

Reply:
And that's not really supported in CDK v1 anyway

> If you want to include headers in the cache key, use a cache policy. For more information, see [Creating cache policies](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/controlling-the-cache-key.html#cache-key-create-cache-policy) in the Amazon CloudFront Developer Guide .

https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_cloudfront.CfnDistribution.ForwardedValuesProperty.html#headers

Reply:
https://github.com/aws/aws-cdk/issues/10829